### PR TITLE
Add functions vcx_credential_get_attributes, vcx_disclosed_proof_get_proof_request_attachment

### PR DESCRIPTION
**Pull request by mirgee**
Signed-off-by: Miroslav Kovar <miroslavkovar@protonmail.com>

### CODE REVIEW

File: libvcx/src/aries/handlers/issuance/holder/states/finished.rs [hunk 1]
```diff
@@ -8,3 +10,25 @@ pub struct FinishedHolderState {
...
 }
+
+impl FinishedHolderState {
+    pub fn get_attributes(&self) -> VcxResult<String> {
+        let credential = self.credential.as_ref().ok_or(VcxError::from_msg(VcxErrorKind::InvalidState, "No credential found"))?;
+        let content = credential.credentials_attach.content()?;
+        let cred_data: CredentialData = serde_json::from_str(&content)
+            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize {:?}, into CredentialData, err: {:?}", content, err)))?;
+
+        let mut new_map: HashMap<String, String> = HashMap::new();
```
Comment (Patrik-Stas) [inline libvcx/src/aries/handlers/issuance/holder/states/finished.rs:21]: Thought this snippet bellow could look bit different. Came up with this:
```
        let mut new_map = serde_json::map::Map::new();
        match cred_data.values.as_object() {
            Some(values) => {
                for (key, value) in values {
                    let val = value["raw"]
                        .as_str()
                        .ok_or(VcxError::from_msg(VcxErrorKind::InvalidJson, "Missing raw encoding on credential value"))?
                        .into();
                    new_map.insert(key.clone(), val);
                };
                Ok(Value::Object(new_map).to_string())
            }
            _ => Err(VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot convert {:?} into object", content)))
        }
```

File: libvcx/src/aries/handlers/issuance/holder/states/offer_received.rs [hunk 2]
```diff
@@ -40,3 +33,20 @@ impl From<(OfferReceivedState, ProblemReport)> for FinishedHolderState {
...
+
+impl OfferReceivedState {
+    pub fn new(offer: CredentialOffer) -> Self {
+        OfferReceivedState {
+            offer,
+        }
+    }
+
+    pub fn get_attributes(&self) -> VcxResult<String> {
+        let mut new_map: HashMap<String, String> = HashMap::new();
```
Comment (Patrik-Stas) [inline libvcx/src/aries/handlers/issuance/holder/states/offer_received.rs:45]: You can use `let mut new_map = serde_json::map::Map::new();` and then return `Ok(Value::Object(new_map).to_string())` online `49` - so there wouldn't be any chance of serialization error on rust typing level. 
(The `insert` method on `serde_json::map::Map` accepts as value only `Value`, which is known to be serializable.)

File: libvcx/src/aries/handlers/issuance/holder/state_machine.rs [hunk 3]
```diff
@@ -223,6 +223,14 @@ impl HolderSM {
         }
     }
 
+    pub fn get_attributes(&self) -> VcxResult<String> {
+        match self.state {
+            HolderState::Finished(ref state) => state.get_attributes(),
+            HolderState::OfferReceived(ref state) => state.get_attributes(),
+            _ => Err(VcxError::from_msg(VcxErrorKind::NotReady, "Cannot get credential attributes: credential offer must be receieved first"))
```
Comment (Patrik-Stas) [inline libvcx/src/aries/handlers/issuance/holder/state_machine.rs:230]: `...credential offer or credential ...`

File: libvcx/src/aries/messages/issuance/credential.rs [hunk 4]
```diff
@@ -35,6 +53,16 @@ impl Credential {
     }
 }
 
+impl TryFrom<A2AMessage> for Credential {
```
Comment (Patrik-Stas) [inline libvcx/src/aries/messages/issuance/credential.rs:56]: Is this used somewhere?

**Comment by Patrik-Stas** [comment]
Can you also extend Java and iOS wrapper? Probably any mobile app would want to use this.

**Review by Patrik-Stas — CHANGES_REQUESTED** [review]
Can you also extend Java and iOS wrapper? Probably any mobile app would want to use this.

File: libvcx/src/aries/handlers/issuance/holder/state_machine.rs (hunk 3 above)
Comment (mirgee) [reply libvcx/src/aries/handlers/issuance/holder/state_machine.rs:230]: The second implies the first :)

**Review by Patrik-Stas — CHANGES_REQUESTED** [review]
please rename added function for proof to getProofRequestAttachment as we agreed

**Review by Patrik-Stas — APPROVED** [review]

**Pull request merged**